### PR TITLE
Add dependencies in the setup.py

### PR DESCRIPTION
**Pull request by saimn**
See #712, cc @astrofrog

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -91,6 +91,12 @@ def run_tests(self):
       author='Chris Beaumont, Thomas Robitaille',
       author_email='glueviz@gmail.com',
       url='http://glueviz.org',
+      install_requires=['numpy', 'pandas', 'astropy', 'matplotlib'],
+      extras_require={
```
Comment (astrofrog) [inline setup.py]: I think it should be ``extras_requires``?

Comment (saimn) [inline setup.py]: It would be more consistent indeed but no, it is `extras_require` :
https://pythonhosted.org/setuptools/setuptools.html#declaring-extras

**Comment by saimn** [comment]
Weird, one of the build is failing with a missing scipy, which is listed in CONDA_DEPS so should have been installed I guess.

**Comment by astrofrog** [comment]
The Travis build that is failing is one which has a minimum of dependencies. For now, the quick solution is to move astropy to be in the ``extra_requires`` section.

**Comment by saimn** [comment]
@astrofrog I added the `requires_scipy` decorator to the two class that need it, and rebased on master, but there is still one test failing on osx.

**Comment by astrofrog** [comment]
@saimn - I have made fixes to the continuous integration testing. Could you rebase on the latest upstream master? (should be trivial, with no conflicts)

**Comment by saimn** [comment]
@astrofrog done !

**Comment by astrofrog** [comment]
Thanks!

**Pull request merged**